specification/devopsinfrastructure/resource-manager/Microsoft.DevOpsInfrastructure/preview/2024-04-04-preview/devopsinfrastructure.json

```diff
@@ -0,0 +1,1858 @@
+            "value": "Allocated",
+            "description": "Represents a machine resource that is allocated."
+          },
+          {
+            "name": "PendingReturn",
+            "value": "PendingReturn",
+            "description": "Represents a machine resource that is pending return."
+          },
+          {
+            "name": "Returned",
+            "value": "Returned",
+            "description": "Represents a machine resource that is returned."
+          },
+          {
+            "name": "Leased",
+            "value": "Leased",
+            "description": "Represents a machine resource that is leased."
+          },
+          {
+            "name": "Provisioning",
+            "value": "Provisioning",
+            "description": "Represents a machine resource that is provisioning."
+          },
+          {
+            "name": "Updating",
+            "value": "Updating",
+            "description": "Represents a machine resource that is updating."
+          },
+          {
+            "name": "Starting",
+            "value": "Starting",
+            "description": "Represents a machine resource that is starting."
+          },
+          {
+            "name": "PendingReimage",
+            "value": "PendingReimage",
+            "description": "Represents a machine resource that is pending reimage."
+          },
+          {
+            "name": "Reimaging",
+            "value": "Reimaging",
+            "description": "Represents a machine resource that is reimaging."
+          }
+        ]
+      }
+    },
+    "SecretsManagementSettings": {
+      "type": "object",
+      "description": "The secret management settings of the machines in the pool.",
+      "properties": {
+        "certificateStoreLocation": {
+          "type": "string",
+          "description": "Where to store certificates on the machine."
+        },
+        "observedCertificates": {
+          "type": "array",
+          "description": "The list of certificates to install on all machines in the pool.",
+          "items": {
+            "type": "string",
+            "format": "uri"
+          }
+        },
+        "keyExportable": {
+          "type": "boolean",
+          "description": "Defines if the key of the certificates should be exportable."
+        }
+      },
+      "required": [
+        "observedCertificates",
+        "keyExportable"
+      ]
+    },
+    "Stateful": {
+      "type": "object",
+      "description": "Stateful profile meaning that the machines will be returned to the pool after running a job.",
+      "properties": {
+        "maxAgentLifetime": {
+          "type": "string",
+          "description": "How long should stateful machines be kept around. The maximum is one week."
+        },
+        "gracePeriod": {
+          "type": "string",
+          "description": "How long should the machine be kept around after it ran a workload when there are no stand-by agents. The maximum is one week."
```

Comment:
Consider using the units in the propertyname 

gracePeriodInHours or graceperiodTmeSpan etc.. This is not really clear what I would enter for say 2 days.